src/container/Login.container.jsx

```diff
@@ -0,0 +1,72 @@
+import {useCallback, useState} from "react";
+
+import {getToken, setAuthTokenCookie} from "../tool/auth.util";
+
+import Login from "../page/Login";
+
+import PropTypes from "prop-types";
+
+const LoginContainer = ({setToken}) => {
+
+    const [username, setUsername] = useState({current: '', error: false});
+    const [password, setPassword] = useState({current: '', error: false});
+    const [error, setError] = useState('');
+
+    const handleSubmit = async e => {
+
+        e.preventDefault();
+
+        if (username.current === undefined || username.current.length === 0) {
+            setUsername({current: username.current, error: true});
+            return;
+        }
+
+        if (password.current === undefined || password.current.length === 0) {
+            setPassword({current: password.current, error: true});
+            return;
+        }
+
+        const response = await getToken(username.current, password.current);
+
+        if (!response.success) {
+            if (response.status === 404 || response.status === 403 || response.status === 401) {
+                setError("Invalid username/password!");
+            } else {
+                setError("Unknown server error!");
+            }
+            return;
+        }
+
+        const token = response.data['token'];
+        if (token === undefined) {
+            setError("Unknown server error!");
+            return;
+        }
+
+        setToken(token);
+        setAuthTokenCookie(token)
+    }
+
+    const onUsernameChange = useCallback((value) => {
+        setError('');
+        setUsername({current: value, error: false});
+    }, []);
+
+    const onPasswordChange = useCallback((value) => {
+        setError('');
+        setPassword({current: value, error: false});
+    }, []);
```

Comment:
So another way to further improve this is if you have a form state object that holds the username and password values as fields, you can then create a single on change function that can consume the field name and the value and you can perform a single on change like something below, where the `newState` value represents something like `{ username: 'exampleusername' }` 
```suggestion
    const onChange = (newState) => {
        setForm({ ...form, ...newState });
    };
```